app/controllers/api/chat_sessions_controller.rb

```diff
@@ -0,0 +1,85 @@
+module Api
+  class ChatSessionsController < BaseController
+    before_action :authenticate_user!, except: [:create, :send_message]
+    before_action :doorkeeper_authorize!, only: [:create, :send_message]
+    before_action :set_chat_session, only: [:retrieve_chat_history, :send_message]
+    before_action :authorize_retrieve_chat_history, only: [:retrieve_chat_history]
+
+    def create
+      bid_item_id = params[:bid_item_id]
+      user_id = current_resource_owner.id
+
+      unless bid_item_id.present? && bid_item_id.to_s.match?(/\A\d+\z/)
+        return render json: { message: I18n.t('common.errors.invalid_bid_item_id') }, status: :bad_request
+      end
+
+      begin
+        chat_session = ChatSessionService.create_chat_session(bid_item_id: bid_item_id.to_i, user_id: user_id)
+        render 'api/chat_sessions/create', locals: { chat_session: chat_session }, status: :created
+      rescue ActiveRecord::RecordNotFound => e
+        render json: { message: e.message }, status: :not_found
+      rescue Exceptions::AuthenticationError, Pundit::NotAuthorizedError => e
+        render json: { message: e.message }, status: :unauthorized
+      rescue StandardError => e
+        render json: error_response(nil, e), status: :internal_server_error
+      end
+    end
+
+    def retrieve_chat_history
+      if @chat_session.is_active
+        messages = @chat_session.messages.includes(:user).page(params[:page]).per(params[:per_page])
+        render 'api/chat_sessions/retrieve_chat_history', locals: { messages: messages }, status: :ok
+      else
+        render json: { message: I18n.t('chat_sessions.chat_disabled') }, status: :forbidden
```

Comment:
it looks starting using localize but still link incorrect and missing key defination @zrg-team